### PR TITLE
Separating documents from the reducer

### PR DESCRIPTION
**Pull request by tejans24**
connects #2829

### CODE REVIEW

File: client/app/reader/PdfListView.jsx [hunk 1]
```diff
@@ -23,6 +23,11 @@ export class PdfListView extends React.Component {
       } else {
         this.props.fetchAppealDetails(this.props.match.params.vacolsId);
       }
+
+      // if appeal is loaded from the assignments and it matches the vacols id
+      // in the url
+    } else if (this.props.appeal.vacols_id === this.props.match.params.vacolsId) {
```
Comment (tejans24) [inline client/app/reader/PdfListView.jsx]: this is a bug fix not to make a call to fetch the appeal details if it already exists in the assignments.

Comment (mdbenjam) [reply client/app/reader/PdfListView.jsx]: Nice I just fixed a similar bug with loading the case. Eventually want to try and move this stuff to an action.

File: client/app/reader/reducer.js [hunk 2]
```diff
@@ -859,50 +853,38 @@ export const reducer = (state = initialState, action = {}) => {
...
-    return updateDocuments(state, action);
+    });
   default:
     return state;
   }
 };
 
+export const readerReducer = (state = initialState, action = {}) => ({
+  ...reducer(state, action),
+  documents: documentsReducer(state.documents, action)
```
Comment (mdbenjam) [inline client/app/reader/reducer.js]: Any reason we do this rather than combine reducers?

Comment (tejans24) [reply client/app/reader/reducer.js]: I am working on another PR that is moving annotations into a separate object outside of the ReaderReducer. I am planning on moving the documents into a separate object as well, right after. That PR has groundwork to make it easier for this change.

**Pull request merged**